### PR TITLE
Docker: build from local code

### PR DESCRIPTION
**Pull request by tyzbit**
This PR slightly adjusts the main Dockerfile to match the one in the docker folder to build from local code rather than pulling from git.

Currently if automated builds are configured, the `latest` tag will always be what's in the repository but if a tag is created on the main repo, when it builds on Docker Hub it will just pull in the latest code in the Dockerfile.

TL;DR This allows automated tagged builds on Docker Hub.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,11 +4,13 @@ FROM golang:alpine as builder
 # queries required to connect to linked containers succeed.
 ENV GODEBUG netdns=cgo
 
+# Copy in the local repository to build from.
+COPY . /go/src/github.com/lightningnetwork/lnd
```
Comment (Roasbeef) [inline Dockerfile]: Wasn't the goal of this container that it _did_ pull from git? We already have the "dev" container that'll copy from the local `lnd` instance.

Comment (tyzbit) [reply Dockerfile]: It was in the beginning, but after thinking about it I think being able to build tagged Docker images  (`lightninglabs/lnd:0.4-beta`, `lightninglabs/lnd:0.4.1-beta` etc etc) would be functionality people would appreciate. Currently, if you tag a commit older than the most current in master, on Docker Hub it will attempt to build but just pull in the latest master (if you have automated builds).

TL;DR building from local code allows creating a tagged commit on GitHub and it automatically generating an appropriately tagged Docker image.

I do concede that this means people who pull down the repo and build their Docker image locally won't get the latest code, but maybe that's a good thing.

**Comment by jlamur** [comment]
@Roasbeef Could you please merge this and setup automated builds on dockerhub?

**Comment by halseth** [comment]
@tyzbit still relevant?

cc @guggero

**Comment by tyzbit** [comment]
Yes, this PR is still relevant IMO.  When you clone a git repo you expect builds from the repo to use the code in the repo.  That is not the case today without this PR.  Additionally, building images on Docker Hub would make LND more accessible to people, and add signal to the noise that's already on there by having an "official" image.

**Comment by guggero** [comment]
The "main" docker file (in the root directory) now allows specifying a specific commit or branch, which allows for automated builds on Dockerhub with the `checkout` build argument.
If you want to build from source, you can use the "dev" docker file that is located in `docker/lnd`, as @roasbeef pointed out.
